### ISSUE TITLE
CUDA should copy ARGV instead of just the pointer

### ISSUE THREAD

**Issue by max-ruttenberg** [bespoke-silicon-group/bsg_f1#283]
https://github.com/bespoke-silicon-group/bsg_f1/blob/2f9fc8460f8566e70b6a616647696eba93da6e8e/cl_manycore/libraries/bsg_manycore_cuda.cpp#L444

In this line of code CUDA stores a pointer to argv instead of making a copy.
The problem with this is consider this CUDA client:
```c
void enqueue_job()
{
    uint32_t argv[] = {0,1,2,3};
    int argc = array_size(argv);
    hb_mc_grid_init(device, /* etc... */, "my_function", argc, argv);
}
void run_job()
{
     hb_mc_device_tile_groups_execute(device);
}
``` 
As is, this code will actually break! 
Because only a pointer to `argv` is saved (as opposed to a deep copy) by the time `run_job()` is called `argv` will have gone out of scope and the pointer that was saved will be invalid.   

Given that `argv` will likely be small making a copy shouldn't really be a big deal.

**Comment by bornaehsani** [comment]
Ahhh so true! You have a good eye for bugs! I actually encountered this problem with large argv lists but never realized what the problem was. I will fix it, thanks for pointing it out!

**Comment by max-ruttenberg** [comment]
If this could be address quickly it would be very helpful.

**Comment by bornaehsani** [comment]
Sorry for the late response! addressed in PR #293 .
